src/main/java/com/google/api/codegen/transformer/ApiCallableTransformer.java

```diff
@@ -0,0 +1,159 @@
+import com.google.api.codegen.viewmodel.BundlingConfigView;
+import com.google.api.tools.framework.model.Method;
+import com.google.api.tools.framework.model.TypeRef;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+public class ApiCallableTransformer {
+
+  public List<ApiCallableView> generateStaticApiCallables(SurfaceTransformerContext context) {
+    List<ApiCallableView> callableMembers = new ArrayList<>();
+
+    for (Method method : context.getInterface().getMethods()) {
+      callableMembers.addAll(generateStaticApiCallables(context.asMethodContext(method)));
+    }
+
+    return callableMembers;
+  }
+
+  public List<ApiCallSettingsView> generateCallSettings(SurfaceTransformerContext context) {
+    List<ApiCallSettingsView> settingsMembers = new ArrayList<>();
+
+    for (Method method : context.getInterface().getMethods()) {
+      settingsMembers.addAll(generateApiCallableSettings(context.asMethodContext(method)));
+    }
+
+    return settingsMembers;
+  }
+
+  private List<ApiCallableView> generateStaticApiCallables(MethodTransformerContext context) {
+    ModelTypeTable typeTable = context.getTypeTable();
+    Method method = context.getMethod();
+    MethodConfig methodConfig = context.getMethodConfig();
+
+    List<ApiCallableView> apiCallables = new ArrayList<>();
+
+    ApiCallableView.Builder apiCallableBuilder = ApiCallableView.newBuilder();
+
+    apiCallableBuilder.requestTypeName(typeTable.getAndSaveNicknameFor(method.getInputType()));
+    apiCallableBuilder.responseTypeName(typeTable.getAndSaveNicknameFor(method.getOutputType()));
+    apiCallableBuilder.name(context.getNamer().getCallableName(method));
+    apiCallableBuilder.settingsFunctionName(context.getNamer().getSettingsFunctionName(method));
+
+    if (methodConfig.isBundling()) {
+      apiCallableBuilder.type(ApiCallableType.BundlingApiCallable);
+    } else {
+      apiCallableBuilder.type(ApiCallableType.SimpleApiCallable);
+    }
+
+    apiCallables.add(apiCallableBuilder.build());
+
+    if (methodConfig.isPageStreaming()) {
+      PageStreamingConfig pageStreaming = methodConfig.getPageStreaming();
+
+      ApiCallableView.Builder pagedApiCallableBuilder = ApiCallableView.newBuilder();
+
+      String pagedResponseTypeName =
+          context
+              .getNamer()
+              .getAndSavePagedResponseTypeName(
+                  typeTable, pageStreaming.getResourcesField().getType());
+
+      pagedApiCallableBuilder.requestTypeName(
+          typeTable.getAndSaveNicknameFor(method.getInputType()));
+      pagedApiCallableBuilder.responseTypeName(pagedResponseTypeName);
+      pagedApiCallableBuilder.name(context.getNamer().getPagedCallableName(method));
+      pagedApiCallableBuilder.settingsFunctionName(
+          context.getNamer().getSettingsFunctionName(method));
+
+      apiCallables.add(pagedApiCallableBuilder.type(ApiCallableType.PagedApiCallable).build());
```

Comment:
nit, very minor - why use the builder's fluent api only in this last line? Maybe either put pagedApiCallableBuilder.type(...) on its own line, or use the fluent api for all the calls to the builder?